src/dotnet-ildasm/Disassembler.cs

```diff
@@ -136,7 +138,8 @@ private void WriteTypeSignature(TypeDefinition type)
             //TODO: Signature to use IL types #2
             _outputWriter.Write($" {type.FullName}");
             //TODO: External Types should always be preceded by their assembly names #6
-            _outputWriter.Write($" extends {type.BaseType.FullName}");
+            if(!type.IsInterface)
```

Comment:
Good point. I made some amendments to add support to Interfaces:
```
if (type.BaseType != null)
    _outputWriter.Write($" extends {type.BaseType.FullName}");

if (type.HasInterfaces)
  _outputWriter.Write($" implements { string.Join(", ", type.Interfaces.Select(x => x.InterfaceType.FullName )) }");    
```

Just committed it and linked to this issue. Thanks for the help! ;)